nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -528,6 +567,8 @@ in
 
     virtualisation.qemu.drives = mkMerge [
       (mkIf cfg.useBootLoader [
+        # The order of this list determines the device names, see
```

Comment:
I fixed this brittle order-determining-device-names thing as part of #72354.  Sorry that I didn't aggressively enough pursue reviewers to get that merged to make this change easier.  I've split out the cleanup and refactor parts into the separate PR #92205 that can hopefully be reviewed and merged faster.

I think it'd be great if we could have fewer "/dev/..." strings flying around in here, and just one `diskInterface == "scsi"
` test about device names.

Consider reviewing #92205?  (Rebasing this atop #92205 would be an excellent way to review #92205!)

Reply:
OK, currently testing the rebase.

Reply:
@chkno With https://github.com/NixOS/nixpkgs/pull/92205#discussion_r449724491 fixed, this PR passes the VM based test I wrote for #85895.

So this is all looking very good, thanks!